drivers/sensor/stm32/temp_stm32.h

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2017 Linaro Limited.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _STM32_TEMP_H_
+#define _STM32_TEMP_H_
+
+#include <stdint.h>
+
+#include <device.h>
+#include <soc.h>
+
+struct temp_stm32_config {
+	ADC_TypeDef *adc;
+	ADC_Common_TypeDef *adc_common;
+	uint8_t adc_channel;
+#if CONFIG_SOC_SERIES_STM32F4X
+	struct stm32f4x_pclken pclken;
+#endif
```

Comment:
Sorry, don't know what you mean -- I'm just following the pattern in e.g. uart_stm32.h:

```
/* device config */
struct uart_stm32_config {
	struct uart_device_config uconf;
	/* clock subsystem driving this peripheral */
#if defined(CONFIG_CLOCK_CONTROL_STM32_CUBE)
	struct stm32_pclken pclken;
#else
#if defined(CONFIG_SOC_SERIES_STM32F1X)
	clock_control_subsys_t clock_subsys;
#elif defined(CONFIG_SOC_SERIES_STM32F4X)
	struct stm32f4x_pclken pclken;
#endif
#endif /* CONFIG_CLOCK_CONTROL_STM32_CUBE */
};
```
I didn't have time to try to make it work on other chips or with another HAL backend.